src/store/homepage/reducer.js

```diff
@@ -0,0 +1,20 @@
+const initialState = {
+  booksCollections: [],
+};
+
+export default function homepageReducer(state = initialState, action) {
+  let newState = { ...state };
+
+  switch (action.type) {
+    case "SET_BOOKS_COLLECTIONS": {
+      newState.booksCollections = action.payload;
+      break;
+    }
+
+    default: {
+      // do nothing
+    }
+  }
+
+  return newState;
+}
```

Comment:
more thoughts on the reducers as i scroll down hahah.
For example, each time you dispatch an action, all reducers get called. Then one of them matches the action and returns something to change the state. 
The way you have it set up, every time an action is dispatch **every** state changes, because you are building a new state object (with all the stuff the old one had yes, but for JS its still a new object). So every selector gets activated again, everything refreshes essentially.

Reply:
Ok, I think, I understand what you mean. I remember, Rein advised to write reducers in 'the usual way', but I started to do  it differently  from the very beginning (just because it was simpler to me). Of course I need to switch to the 'correct' way. Btw. I expected the comment about my reducers...